server/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -379,4 +383,36 @@ public Result destroyEligibility(long programId, long blockDefinitionId) {
                 programId, blockDefinitionId))
         .flashing(FlashKey.SUCCESS, "Removed the eligibility condition for this screen.");
   }
+
+  /** POST endpoint for updating eligibility message. */
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result updateEligibilityMessage(Request request, long programId, long blockDefinitionId) {
+    requestChecker.throwIfProgramNotDraft(programId);
+
+    Form<BlockEligibilityMessageForm> EligibilityMsgform =
+        formFactory.form(BlockEligibilityMessageForm.class).bindFromRequest(request);
+    String newMessage = EligibilityMsgform.get().getEligibilityMessage();
+    String toastMessage;
+    String toastType;
+
+    try {
+      programService.setBlockEligibilityMessage(
+          programId, blockDefinitionId, Optional.of(LocalizedStrings.of(Locale.US, newMessage)));
+
+      toastType = "success";
+      if (newMessage.isBlank()) {
+        toastMessage = "Eligibility message removed.";
+      } else {
+        toastMessage = "Eligibility message set to " + newMessage;
```

Comment:
When the message is very long and includes markdown, what happens to the toast?

Reply:
Good question, I'll test that scenario in browser tests when implementing frontend.